modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java

```diff
@@ -104,6 +239,55 @@ public void testGetOpenAPI() throws Exception {
 		Assert.assertEquals("NOT_FOUND", jsonObject.getString("status"));
 	}
 
+	private String _getNestedEntitySchema(
+		JSONObject jsonObject, ObjectRelationship objectRelationship,
+		ObjectDefinition objectDefinition) {
+
+		String nestedEntitySchema;
+
+		if (Objects.equals(
+				objectRelationship.getType(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+			(objectDefinition.getObjectDefinitionId() ==
+				_objectDefinition2.getObjectDefinitionId())) {
+
+			nestedEntitySchema = (String)jsonObject.getJSONObject(
+				"components"
+			).getJSONObject(
+				"schemas"
+			).getJSONObject(
+				objectDefinition.getShortName()
+			).getJSONObject(
+				"properties"
+			).getJSONObject(
+				objectRelationship.getName()
+			).get(
+				"$ref"
+			);
+		}
+		else {
+			nestedEntitySchema = (String)jsonObject.getJSONObject(
```

Comment:
The way that we are handling the nestedEntitySchema looks equal to me except for one JSONObject. Could we avoid duplicate code?